cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/generic/DynamicModel.java

```diff
@@ -49,6 +56,17 @@ public <T> T getParameter(String key, Class<T> clazz) {
         return clazz.cast(parameters.get(key));
     }
 
+    public <T> T getParameterWithFallback(String key, Class<T> clazz) {
+        try {
+            return clazz.cast(parameters.get(key));
+        } catch (ClassCastException e) {
+            LOGGER.error("Can't cast to {}, trying to read it as an Object, then write it to json and try to read it to {}", clazz, clazz);
+            Object object = getParameter(CloudResource.ATTRIBUTES, Object.class);
```

Comment:
Should we not use the `key` variable here instead of `CloudResource.ATTRIBUTES`?

Reply:
very good catch, thank you @Bajzathd !